src/geometric_sampler.ml

```diff
@@ -0,0 +1,34 @@
+type t =
+  { rand : Random.State.t;
+    one_log1m_lambda : float }
+
+let default_rand () =
+  Random.State.make [| 0x52b87efb; 0x332235ea; 0x5f813723; 0x057b9dff |]
+
+let make ?(rand = default_rand ()) ~sampling_rate () =
+  let one_log1m_lambda =
+    if sampling_rate >= 1. then 0. else 1./.log1p(-. sampling_rate) in
```

Comment:
It's not meaningful to have sampling_rate > 1, but I wanted to avoid blowing up. (I'm imagining a case where someone computes a sampling rate, and ends up with 1.000000001 because of rounding error).